pkg/util/json/json.go

```diff
@@ -701,6 +705,23 @@ func (j jsonObject) Exists(s string) bool {
 	return j.FetchValKey(s) != nil
 }
 
+func (jsonNull) ObjectKeyIterator() <-chan JSON   { return nil }
+func (jsonTrue) ObjectKeyIterator() <-chan JSON   { return nil }
+func (jsonFalse) ObjectKeyIterator() <-chan JSON  { return nil }
+func (jsonNumber) ObjectKeyIterator() <-chan JSON { return nil }
+func (jsonString) ObjectKeyIterator() <-chan JSON { return nil }
+func (jsonArray) ObjectKeyIterator() <-chan JSON  { return nil }
+func (j jsonObject) ObjectKeyIterator() <-chan JSON {
+	ch := make(chan JSON)
```

Comment:
This is cool, and very simple! But I have a couple anxieties about using channels and async code where sync code will suffice, besides a general fear of raceyness: firstly, I think they're pretty slow in general (though I'd have to benchmark to tell you exactly how slow), since every pull/push will have to go through the go runtime scheduler, but secondly and a little more concerning is that it's not completely clear to me if the generators will always be sure to completely exhaust their values, meaning if somehow someone only accesses a subset of the object values, we could end up with this channel never getting closed and this goroutine living forever (after writing this I see that the generators have a `Close` method which we could possibly use, but I'm still a little nervous).

It's a little more syntactically loud, but I much prefer a pattern like:
```
type ObjectKeyIterator struct {
  source jsonObject
  idx int
}
// Next returns true and the next key in the iterator if one exists, and false otherwise.
func (o *ObjectKeyIterator) Next() (more bool, key string) {
...
}
```

And that way we're safe because if we lose a handle onto the iterator then it will still just get garbage collected eventually.

What do you think?

Reply:
You're right. It is a bit dangerous to expose the channel outside the function.
I'll implement it with iterator pattern. Thank you!